wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -221,32 +128,44 @@ public static void main(String... args) throws Exception {
     return null;
   }
 
+  @Deprecated
   public WireCompiler(String protoPath, List<String> sourceFileNames, List<String> roots,
```

Comment:
That would be awesome. I checked the maven plugin and it ends up calling WireCompiler.main() with a note that we need a better interface to invoke the compiler.